lib/vrl/stdlib/src/parse_aws_cloudtrail_logs.rs

```diff
@@ -0,0 +1,1199 @@
+enum AwsCloudTrailLogsRecordUserIdentityType {
+    Root,
+    #[serde(rename(deserialize = "IAMUser"))]
+    IamUser,
+    AssumedRole,
+    FederatedUser,
+    #[serde(rename(deserialize = "AWSAccount"))]
+    AwsAccount,
+    #[serde(rename(deserialize = "AWSService"))]
+    AwsService,
+}
+
+impl From<AwsCloudTrailLogsRecordUserIdentityType> for Value {
+    fn from(identity_type: AwsCloudTrailLogsRecordUserIdentityType) -> Self {
+        format!("{}", identity_type).into()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct AwsCloudTrailLogsRecordUserIdentitySessionContext {
+    session_issuer: Option<AwsCloudTrailLogsRecordUserIdentitySessionContextSessionIssuer>,
+    web_id_federation_data:
+        Option<AwsCloudTrailLogsRecordUserIdentitySessionContextWebIdFederationData>,
+    attributes: Option<BTreeMap<String, Value>>,
+}
+
+impl From<AwsCloudTrailLogsRecordUserIdentitySessionContext> for Value {
+    fn from(session_context: AwsCloudTrailLogsRecordUserIdentitySessionContext) -> Self {
+        let mut value = BTreeMap::<String, Value>::new();
+
+        if let Some(session_issuer) = session_context.session_issuer {
+            value.insert("session_issuer".to_owned(), session_issuer.into());
+        }
+        if let Some(web_id_federation_data) = session_context.web_id_federation_data {
+            value.insert(
+                "web_id_federation_data".to_owned(),
+                web_id_federation_data.into(),
+            );
+        }
+        if let Some(attributes) = session_context.attributes {
+            value.insert("attributes".to_owned(), attributes.into());
+        }
+
+        value.into()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct AwsCloudTrailLogsRecordUserIdentitySessionContextSessionIssuer {
+    r#type: AwsCloudTrailLogsRecordUserIdentitySessionContextSessionIssuerType,
+    user_name: String,
+    principal_id: String,
+    arn: String,
+    account_id: String,
+}
+
+impl From<AwsCloudTrailLogsRecordUserIdentitySessionContextSessionIssuer> for Value {
+    fn from(
+        session_issuer: AwsCloudTrailLogsRecordUserIdentitySessionContextSessionIssuer,
+    ) -> Self {
+        let mut value = BTreeMap::<String, Value>::new();
+
+        value.insert("type".to_owned(), session_issuer.r#type.into());
+        value.insert("user_name".to_owned(), session_issuer.user_name.into());
+        value.insert(
+            "principal_id".to_owned(),
+            session_issuer.principal_id.into(),
+        );
+        value.insert("arn".to_owned(), session_issuer.arn.into());
+        value.insert("account_id".to_owned(), session_issuer.account_id.into());
+
+        value.into()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct AwsCloudTrailLogsRecordUserIdentitySessionContextWebIdFederationData {
```

Comment:
I think since this struct is private it would be ok to drop the `AwsCloudTrailLogs` prefix. Although maybe there is a prize for the longest struct name in the Rust ecosystem, so I understand if you want to keep it!

Reply:
Well... 😆 The only thing I can say here is that I'd love nested struct declarations in Rust to keep the namespace clean!

Reply:
I'm getting Enterprise Java(TM) flashbacks